### ISSUE TITLE
Do not fail parsing config on invalid keys or steps

### ISSUE THREAD

**Issue by xeruf** [topgrade-rs/topgrade#491]
## Erroneous Behavior
Topgrade ignores the config when it detects an unknown step or config key.

## Expected Behavior
It warns but uses the remainder of the config.

## Steps to reproduce
Run an older version of topgrade with a new config file (e.g. when using a config file on Debian and Arch through a dotfiles repo)

See https://github.com/r-darwish/topgrade/issues/847

**Comment by SteveLauC** [comment]
This is not something hard to implement, removing `#[serde(deny_unknown_fields)]` from types in `src/config.rs` would get it done.

But this will make users hard to find mistakes in their configuration file...

**Comment by xeruf** [comment]
Just spit out a warning but use the config, as I outlined as expectation above.
How would that hinder users from finding mistakes in their config?

Once again today topgrade ignores all my config and performs wrong steps because some parameter change in an update...

**Comment by SteveLauC** [comment]
> Once again today topgrade ignores all my config and performs wrong steps because some parameter change in an update...

Hi, I am still not convinced this is a good feature to have, what about updating your configuration to the latest one?

And, modifications to the configuration will be seen as breaking changes, after Topgrade 14.0.0, they will be printed on your first run:

```
$ topgrade --version
Topgrade 14.0.0

$ topgrade

── 11:16:41 - Topgrade 14.0.0 Breaking Changes ─────────────────────────────────
1. In 13.0.0, we introduced a new feature, pushing git repos, now this feature
   has been removed as some users are not satisfied with it.

   For configuration entries, the following ones are gone:

  
   [git]
   pull_only_repos = []
   push_only_repos = []
   pull_arguments = ""
   push_arguments = ""
  


Confirmed? (y)es/(N)o
```

so it won't be hard for users to find and correct them:>

**Comment by xeruf** [comment]
I ust topgrade across diferent operating systems, such as arch and debian, which might be published in different versions

I might add a config flag for the newer version on arch, sync my dotfiles to a debian machine, and then topgrade won't work until debian has the newest package

so it's not about an incorrect flag, just a not-yet-implemented one

**Comment by GideonBear** [comment]
@xeruf Good point. Would failing be a better solution than carrying on with default config?

**Comment by GideonBear** [comment]
Re-opened this for more discussion, this does not mean it is accepted, or will not be closed again in the future
